saleor/graphql/shop/mutations.py

```diff
@@ -288,3 +288,92 @@ def perform_mutation(cls, _root, info, key_type):
 
         instance.delete()
         return AuthorizationKeyDelete(authorization_key=instance, shop=Shop())
+
+
+class StaffNotificationRecipientInput(graphene.InputObjectType):
+    user = graphene.ID(required=False, description="The ID of the user to notify.")
+    staff_email = graphene.String(
+        required=False, description="Email to notify instead of given staff member."
+    )
+    active = graphene.Boolean(
+        required=False, description="Is staff notification recipient active."
+    )
+
+
+class StaffNotificationRecipientCreate(ModelMutation):
+    class Arguments:
+        input = StaffNotificationRecipientInput(
+            required=True,
+            description="Fields required to create a staff notification recipient.",
+        )
+
+    class Meta:
+        description = "Creates a new staff notification recipient."
+        model = account_models.StaffNotificationRecipient
+        permissions = ("site.manage_settings",)
+        error_type_class = ShopError
+        error_type_field = "shop_errors"
+
+    @classmethod
+    def clean_input(cls, info, instance, data):
+        cleaned_input = super().clean_input(info, instance, data)
+        staff_email = cleaned_input.get("staff_email")
+        user = cleaned_input.get("user")
+        cls.validate_input(instance, staff_email, user)
+        if staff_email:
+            staff_user = account_models.User.objects.filter(email=staff_email).first()
+            if staff_user:
+                cleaned_input["user"] = staff_user
+                cleaned_input["staff_email"] = None
+        return cleaned_input
+
+    @staticmethod
+    def validate_input(instance, staff_email, user):
+        if not instance.id and not staff_email and not user:
```

Comment:
Why are we checking the ID is set?

Reply:
Because update also use this method, so to not duplicate code I just check if instance is updated or not.
And why I'd and not 'if instance' ? Because 'clean_input' pass empty 'instance' 
That's why 'if instance' won't work

Reply:
Only issue there, is that one could set user or the email to `None`, and the validation will accept it despite being invalid; so the `get` is actually not valid in this context (my bad 👀). In that case, we should check whether the user passed those variables or not.

Reply:
added additional check